src/Microsoft.Data/ArrowStringColumn.cs

```diff
@@ -0,0 +1,348 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Text;
+using Apache.Arrow;
+using Apache.Arrow.Types;
+
+namespace Microsoft.Data
+{
+    /// <summary>
+    /// An immutable to hold Arrow style strings
+    /// </summary>
+    public partial class ArrowStringColumn : BaseColumn
```

Comment:
I'm not sure I like using the name `ArrowStringColumn`. My first inclination was to call it `UTF8StringColumn` or similar.

However, I'm not sure I like my suggestion. So I'm just raising this thought here to see if you have any ideas on what could be a better name.

Reply:
Believe me, I spent a while just thinking about the name here and decided to just call it `ArrowStringColumn` for now and rename once we have consensus. My inclination was to call it `ImmutableStringColumn`, but that gives an impression that it is related to the already existing `StringColumn`, when they are not really related. OTOH, it does make the immutability constraint obvious. I actually do like `UTF8StringColumn`. It makes the difference with `StringColumn` obvious.